commands/pull_request.go

```diff
@@ -188,27 +193,32 @@ func pullRequest(cmd *Command, args *Args) {
 	var editor *github.Editor
 	var title, body string
 
+	baseTracking := base
+	headTracking := head
+
+	remote := gitRemoteForProject(baseProject)
+	if remote != nil {
+		baseTracking = fmt.Sprintf("%s/%s", remote.Name, base)
+	}
+	if remote == nil || !baseProject.SameAs(headProject) {
+		remote = gitRemoteForProject(headProject)
+	}
+	if remote != nil {
+		headTracking = fmt.Sprintf("%s/%s", remote.Name, head)
+	}
+
 	if cmd.FlagPassed("message") {
 		title, body = readMsg(flagPullRequestMessage)
 	} else if cmd.FlagPassed("file") {
 		title, body, editor, err = readMsgFromFile(flagPullRequestFile, flagPullRequestEdit, "PULLREQ", "pull request")
 		utils.Check(err)
 	} else if flagPullRequestIssue == "" {
-		baseTracking := base
-		headTracking := head
-
-		remote := gitRemoteForProject(baseProject)
-		if remote != nil {
-			baseTracking = fmt.Sprintf("%s/%s", remote.Name, base)
-		}
-		if remote == nil || !baseProject.SameAs(headProject) {
-			remote = gitRemoteForProject(headProject)
-		}
-		if remote != nil {
-			headTracking = fmt.Sprintf("%s/%s", remote.Name, head)
+		headForMessage := headTracking
+		if flagPullRequestPush {
+			headForMessage = head
```

Comment:
The message is generated before the branch is pushed, so that the user has a chance to back out of pushing by passing an empty message. This means that you want to generate the message based on the local head, since the remote head doesn't have all the relevant commits yet.